tools/slang-test/unit-test-free-list.cpp

```diff
@@ -0,0 +1,55 @@
+// unit-test-free-list.cpp
```

Comment:
Is there any easy way for us to put the actual unit tests under `tests/unit-tests/`? I think that should be reasonably easy if we add that source path to our Premake build for `slang-test`, at which point adding a unit test would require only two steps:

1. Add the source file for your test somewhere under `tests/unit-tests/`
2. Run the premake generator to rebuild the project files

Step (2) there isn't ideal, but I can't come up with an alternative that isn't worse right now.

I'm fine with keeping the unit test source files under `slang-test` for now for simplicity, but it gets to a secondary thing I might want us to consider down the line: I'd like to have a clean split between the test "driver" application that actually loads tests, runs them, and reports results, vs. the tests themselves. I don't think each unit test needs to build its own binary, but maybe they can all build into a DLL that the test driver loads and then runs.

The reasoning for this is really about long-term megalomania for the Slang project, where I'd like `slang-test` to be something we can distribute with the compiler and allow users to easily use it for testing of their own `.slang` files. The tools is *very* far from being able to achieve that goal, so this is not a priority any time soon.

Reply:
I was originally going to make a stand alone exe for running unit tests. But because TestContext needs to be able to use stuff in OS it implies TestContext and OS need to be shared somewhere. core? Probably not. I could create a new lib like gfx... but this all seemed a bit like overkill. 

If you want thought I can split out that way. 

Putting tests in separate directory - sure think this is good idea.

Reply:
The stuff in the "OS" code for the test framework should probably migrate into `core` sooner or later, but that doesn't need to happen today.

As always, my feedback is more about long-term decisions and less about "this must be fixed before commit."